Implement strict Content Security Policy using react-helmet-async

piyushsinghgaur1 opened the pull request:
---

###  **Description:**

This PR enhances the security of the React frontend by implementing a robust Content Security Policy (CSP) using `react-helmet-async`.

### Key Changes:

* Integrated `react-helmet-async` to manage CSP via React components
* Defined a strict CSP (`default-src 'self'`) to block untrusted sources
* Removed `<meta http-equiv="Content-Security-Policy">` from `index.html`
* Whitelisted trusted domains for scripts, styles, fonts, and images
* Validated implementation with browser dev tools — no CSP violations
* Enabled support for resolving nested environment variables using env: placeholders

### Why This Matters:

A strong CSP helps protect against XSS, clickjacking, and code injection attacks by limiting which sources are allowed to load content in the app.

---

piyushsinghgaur1 committed:
feat(core): ARC-78 added CSP

feat(security): implement CSP via react-helmet-async in React frontend

AryanshSourcefuse commented on package.json:
```diff
@@ -36,13 +36,15 @@
     "@tanstack/react-table": "^8.8.5",
     "c8": "^7.12.0",
     "date-fns": "^2.29.3",
+    "dotenv": "^16.5.0",
```
No need to install dotenv
https://vite.dev/guide/env-and-mode

Follow above guide for env integration with vite

piyushsinghgaur1 replied:
Done

Commit:
feat(config): update environment files with CSP directives and client configuration

Add CSP directives and client config to environment files

Updates example environment files to include Content Security Policy
(CSP) directives with secure defaults and client configuration keys.

Introduces logic to dynamically expand environment variables within
CSP directives, validates critical directives, and ensures secure
defaults where missing. Enhances security and configurability.

Relates to: feat(config)

rohit-sourcefuse commented:
No longer required, as we've implemented a generic CSP package for both React and Angular boilerplates. See: https://github.com/sourcefuse/arc-spa-csp
